lib/utils.js

```diff
@@ -624,9 +624,15 @@ exports.populate = function populate(path, select, model, match, options, subPop
 
   var ret = [];
   var paths = path.split(' ');
+  var models = undefined;
+  if (typeof model === 'string') {
+    models = model.split(' ');
+  }
   options = exports.clone(options, { retainKeyOrder: true });
   for (var i = 0; i < paths.length; ++i) {
-    ret.push(new PopulateOptions(paths[i], select, match, options, model, subPopulate));
+    var _model = model;
+    if (models && i < models.length) _model = models[i];
```

Comment:
if the paths length > models length, only models exist and the index in the paths repeat is little than models length , will do "_model = models[i]" , if not , the "_model" as default "model", there is nothing change as before!

if the paths length < models length, only repeat paths length times to do "_model = models[i]", other models is ignore!

I only change the models rule , the other code is as before to do ,
if someone input the illegal models, will get the error as before.